lib/rmt/cli/repos.rb

```diff
@@ -16,6 +16,52 @@ def list
   end
   map 'ls' => :list
 
+  desc 'clean', _('Removes un-mirrored repositories from the local disk.')
+  def clean
+    base_directory = RMT::DEFAULT_MIRROR_DIR
+
+    repos_to_delete = Repository.where(mirroring_enabled: false).map do |repo|
+      repository_dir = File.join(base_directory, repo.local_path)
+      Dir.exist?(repository_dir) ? repo : nil
+    end.compact
+
+    if repos_to_delete.empty?
+      puts _('No un-mirrored repositories found on local disk.')
+      return
+    end
+
+    puts _('RMT found the following un-mirrored repositories:')
+    print "\n\e[31m"
+    repos_to_delete.each do |repo|
+      puts repo.description
+    end
+    print "\n\e[0m\e[1m"
+    print _('Would you like to continue and remove these directories from your local disk?')
+    print "\n\e[22m\s\s"
+    print _("Only '%{input}' will be accepted.") % { input: 'yes' }
+    print "\n\n\s\s\e[1m"
+    print _('Enter a value:')
+    print "\e[22m\s\s"
+
+    input = $stdin.gets.to_s.strip
+    if input != 'yes'
+      puts "\n" + _('Clean cancelled.')
+      return
+    end
+
+    print "\n"
+    repos_to_delete.each do |repo|
+      path = File.join(base_directory, repo.local_path)
+      FileUtils.rm_r(path, secure: true)
+      DownloadedFile.where('local_path LIKE ?', "#{path}%").destroy_all
+      puts _("Deleted repository '%{repo}'.") % { repo: repo.description }
```

Comment:
I'm not sure about the "*deleted*" word here: we delete the repository files and directories, but, if I correctly understood, the repository would still be listed with `rmt-cli repos list`, as "Not mirrored".

In my previous review I wondered about the need to also remove repositories from the database. If this is the case, then probably the command `rmt-cli repos clean` wouldn't make so much sense. Just to make sure we agree on the meaning of **clean**:

1. It will remove all repository files from disk but still keeps the repository as disabled in the database, or;
1. In addition to files on disk, it also removes the repository entry from the database.

If we just want to clean files, maybe we could change to something like this:

```suggestion
      puts _("Deleted files from repository '%{repo}'.") % { repo: repo.description }
```

Or something that better states RMT is just removing files.

Currently, I guess `rmt-cli` don't have an option to remove a repository from the list (database).

Reply:
@lcaparroz You are totally right. We should never delete repositories from the database, only delete their local files.